src/main/java/org/sagebionetworks/web/client/GlobalApplicationStateImpl.java

```diff
@@ -483,61 +487,61 @@ public void initializeToastContainer() {
+        (HTMLDivElement) DomGlobal.document.querySelector("[id=\"dropzone\"]");
+      HTMLDivElement rootPanelElement =
+        (HTMLDivElement) DomGlobal.document.querySelector("[id=\"rootPanel\"]");
+      initializeDragDrop(this, dropZoneElement, rootPanelElement);
     }
   }
 
-  private static final native void _initializeDragDrop(
+  private static void initializeDragDrop(
     GlobalApplicationStateImpl globalAppState,
-    Element dropZone,
-    Element rootPanel
-  ) /*-{
-		try {
-			function showDropZone() {
-				dropZone.style.display = "block";
-			}
-
-			function hideDropZone() {
-				dropZone.style.display = "none";
-			}
-
-			$wnd
-					.addEventListener(
-							'dragenter',
-							function(e) {
-								if (globalAppState.@org.sagebionetworks.web.client.GlobalApplicationStateImpl::isDragAndDropListenerSet()()) {
-									showDropZone();
-								}
-							});
-
-			function allowDrag(e) {
-				e.dataTransfer.dropEffect = 'copy';
-				e.preventDefault();
-			}
-
-			function handleDrop(e) {
-				e.preventDefault();
-				hideDropZone();
-				globalAppState.@org.sagebionetworks.web.client.GlobalApplicationStateImpl::onDrop(Lelemental2/dom/FileList;)(e.dataTransfer.files);
-			}
-
-			dropZone.addEventListener('dragenter', allowDrag);
-			dropZone.addEventListener('dragover', allowDrag);
-
-			dropZone.addEventListener('drop', handleDrop);
-
-			//if files are dropped into the root panel, then ignore the event (do not open file contents if user does not have the upload dialog open).
-			rootPanel.addEventListener('drop', function(e) {
-				e.preventDefault();
-			});
-			rootPanel.addEventListener('dragenter', allowDrag);
-			rootPanel.addEventListener('dragover', allowDrag);
-		} catch (err) {
-			console.error(err);
-		}
-	}-*/;
+    HTMLDivElement dropZone,
+    HTMLDivElement rootPanel
+  ) {
+    try {
+      Runnable showDropZone = () -> dropZone.style.set("display", "block");
+      Runnable hideDropZone = () -> dropZone.style.set("display", "none");
+
+      DomGlobal.window.addEventListener(
+        "dragenter",
+        e -> {
+          if (globalAppState.isDragAndDropListenerSet()) {
+            showDropZone.run();
+          }
+        }
+      );
+
+      EventListener allowDrag = e -> {
+        if (e instanceof DragEvent) {
+          ((DragEvent) e).dataTransfer.dropEffect = "copy";
+          e.preventDefault();
+        }
+      };
+      EventListener handleDrop = e -> {
+        if (e instanceof DragEvent) {
+          e.preventDefault();
+          hideDropZone.run();
+          globalAppState.onDrop(((DragEvent) e).dataTransfer.files);
+        }
+      };
+
+      dropZone.addEventListener("dragenter", allowDrag);
+      dropZone.addEventListener("dragover", allowDrag);
+
+      dropZone.addEventListener("drop", handleDrop);
+      dropZone.addEventListener("dragend", e -> hideDropZone.run());
+      dropZone.addEventListener("dragleave", e -> hideDropZone.run());
```

Comment:
Add dragend/dragleave handlers to fix bug where if you dragged into the window, then dragged out of the window, the dropzone wouldn't go away.